src/core/codegen.rs

```diff
@@ -58,11 +58,12 @@ pub trait CodeGenWrite: Write {
         for c in s.chars() {
             if !is_boundary && c.is_ascii_uppercase() {
                 self.write_char('-')?;
-                self.write_char((c as u8 - b'A' + b'a') as char)?;
+                self.write_char(c.to_ascii_lowercase())?;
+                is_boundary = false;
```

Comment:
in this branch, is_boundry will always be `false`, due to  `c.is_ascii_alphanumeric` is true.